tfexec/internal/e2etest/providers_schema_test.go

```diff
@@ -136,8 +211,9 @@ func TestProvidersSchema(t *testing.T) {
 				}
 
 				expected := c.expected(tfv)
-				if !reflect.DeepEqual(schemas, expected) {
-					t.Fatalf("expected %+v, but got %+v", spew.Sdump(expected), spew.Sdump(schemas))
+
+				if diff := cmp.Diff(expected, schemas, cmp.Exporter(func(_ reflect.Type) bool { return true })); diff != "" {
```

Comment:
I reckon the custom exporter is here because of cty? There is `ctydebug.CmpOptions` https://pkg.go.dev/github.com/zclconf/go-cty-debug@v0.0.0-20191215020915-b22d67c1ba0b/ctydebug#pkg-variables which provides equality comparator and decent output.

Reply:
Yes, the issue here is that the cty types have a number of unexported fields, at paths too awkward to pass to go-cmp's `cmp.AllowUnexported` or `cmpopts.IgnoreUnexported`. Thanks for bringing `ctydebug` to my attention - I'll try that out.